server/storage/consistent_store.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/docker/notary"
+	"github.com/docker/notary/tuf/data"
+)
+
+// ConsistentMetaStorage wraps a MetaStore in order to walk the TUF tree for GetCurrent in a consistent manner
```

Comment:
You're correct in that it's lookup by hash after we retrieve the latest timestamp -- I couldn't think of a great name at the time, I'll try a rename :+1: